### PR TITLE
Enable rewards

### PR DESCRIPTION
**Pull request by kimikowang**
1. Enable adding reward to analytics request
2. Add test for rewards
3. Update analytics path
4. Fix start time in request to analytics

### CODE REVIEW

File: pkg/analytics/service.go [hunk 1]
```diff
@@ -87,18 +88,39 @@ func MakeRequest(instance *iter8v1alpha1.Experiment, baseline, experiment interf
...
+		}
+
+		tc[api.TCKeyReward] = reward
+	}
+
 	return &api.Request{
 		Name: instance.Name,
 		Baseline: api.Window{
-			StartTime: instance.ObjectMeta.GetCreationTimestamp().Format(time.RFC3339),
+			StartTime: time.Unix(0, instance.Status.StartTimestamp).Format(time.RFC3339),
```
Comment (kalantar) [inline pkg/analytics/service.go]: Why is this a better way to compute time? What is the problem?

**Comment by kalantar** [comment]
As part of PRs that rely on PRs from iter8-analytics (I am making an assumption here), could we document what PRs are necessary so that it is easier to run tests?

File: pkg/analytics/service.go (hunk 1 above)
Comment (kimikowang) [reply pkg/analytics/service.go]: Initialization is happening between creation and start time. Only when all targets are detected then the experiment starts.

**Comment by kimikowang** [comment]
> As part of PRs that rely on PRs from iter8-analytics (I am making an assumption here), could we document what PRs are necessary so that it is easier to run tests?

Please test is with this pr: https://github.com/iter8-tools/iter8-analytics/pull/40

File: pkg/analytics/service.go [hunk 2]
```diff
@@ -87,18 +88,39 @@ func MakeRequest(instance *iter8v1alpha1.Experiment, baseline, experiment interf
...
+			api.SCKeyIsCounter:          iter8metric.IsCounter,
+			api.SCKeyAbsentValue:        iter8metric.AbsentValue,
+		}
+
+		if rw.MinMax != nil {
+			reward[api.SCKeyMinMax] = rw.MinMax
+		}
+
+		tc[api.TCKeyReward] = reward
+	}
```
Comment (kalantar) [inline pkg/analytics/service.go]: By placing this in service.go, are we making it applicable to all algorithms? I thought rewards were not valid for check_and_increment.

File: pkg/analytics/service.go (hunk 1 above)
Comment (kalantar) [reply pkg/analytics/service.go]: Ah. I see we've added a create time as well.

File: pkg/analytics/service.go (hunk 2 above)
Comment (kimikowang) [reply pkg/analytics/service.go]: Yes it's applicable to all. We do not do validation on the controller side.

**Comment by kalantar** [comment]
looks ok to merge

**Comment by kalantar** [comment]
@kimikowang is there a reason not to merge this now? the analytics pr that it depends on seems to have been merged

**Pull request merged**